src/Stack/Build.hs

```diff
@@ -128,6 +132,65 @@ allLocal =
     Map.elems .
     planTasks
 
+-- | See https://github.com/commercialhaskell/stack/issues/1198.
+warnIfExecutablesWithSameNameCouldBeOverwritten
+    :: MonadLogger m => [LocalPackage] -> Plan -> m ()
+warnIfExecutablesWithSameNameCouldBeOverwritten locals plan = do
+    let localDuplicateExes = Map.filter (\ps -> length ps > 1) (localExes locals)
```

Comment:
The mixture of let and where is a little confusing to me.  How about moving everything to the where clause, and:

* Inline `exesToBuild`.  The alternative is to take the `Plan` parameter off `plannedExes`, but I like how that scopes the usage of RecordWildCards.

* Remove the `locals'` parameter from `localExes`, and just use `locals`.  Along with this change, may as well move `pkgExePairs` to the where clause

* Possibly combine the definition of localExes and `localDuplicateExes`.  It's fine either way.